regex.c

```diff
@@ -107,6 +142,61 @@ void free_regex(void)
 	{
 		domains[i].regexmatch = REGEX_UNKNOWN;
 	}
+
+	// Also free array of whitelisted domains
+	free_whitelist_domains();
+}
+
+static void read_whitelist_from_file(void)
+{
+	FILE *fp;
+	char *buffer = NULL;
+	size_t size = 0;
+
+	// Get number of lines in the regex file
```

Comment:
These comments still need to be adjusted (they say "regex" instead of "whitelist")